src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -16,8 +16,8 @@ public interface MachineService {
      * @param type - any class of Machine sub class. For example: Truck.class or Bulldozer.class
      * @return the list of machines
      */
-    List<Object> getAll(Class type);
 
+    List<T> getAll(Class<? extends Machine> type);
```

Comment:
The `getAll` method should not allow any type that is not a subtype of Machine. Consider using bounded type parameters to enforce this.